### PR TITLE
PoC [WIP]

### PR DESCRIPTION
**Pull request by TheLuckyOne**
| Q             | A
| ------------- | ---
| Bug fix?      | No
| New feature?  | Yes
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | #1475 

Hi. What do you think about this implementation?

### CODE REVIEW

**Comment by antonmedv** [comment]
What is a few tasks was added?

**Comment by TheLuckyOne** [comment]
I think for multiple tasks in before/after we need group task 
`$task = new GroupTask($name, $body);`

**Comment by antonmedv** [comment]
I think it should not must be unique, otherwise it will rise a lot of questions.

File: src/functions.php [hunk 1]
```diff
@@ -171,40 +171,71 @@ function task($name, $body = null)
...
 
-    $beforeTask->addBefore($that);
+    if (is_callable($that)) {
+        $taskName = 'before_' . $it . '_' . (count($beforeTask->getBefore()) + 1);
+        $task = new T($taskName, $that);
+        $deployer->tasks->set($taskName, $task);
+    } elseif (is_string($that)) {
+        $taskName = $that;
+    } else {
+        throw new \InvalidArgumentException('Task should be an closure or other task.');
```
Comment (staabm) [inline src/functions.php]: other task name

**Comment by TheLuckyOne** [comment]
Guys, tell me how can i improve my PR

**Comment by antonmedv** [comment]
Again, what about adding a few callbacks this way? Really want this feature.

**Comment by TheLuckyOne** [comment]
@antonmedv Feature added in "Added dynamic task name in before/after".

**Comment by staabm** [comment]
sorry I cannot follow your conversation? what exactly is missing?

**Comment by marvinhinz** [comment]
This is actually a very useful feature, what needs to be done to merge this?

**Comment by antonmedv** [comment]
I think it should be rebased.

File: test/src/FunctionsTest.php [hunk 2]
```diff
@@ -131,6 +131,28 @@ public function testBefore()
         self::assertEquals(['before', 'main'], $names);
     }
 
+    public function testCallableBefore()
+    {
+        task('main', 'pwd');
+        before('main', function () {
```
Comment (staabm) [inline test/src/FunctionsTest.php]: you should assert that this callables are actually invoked

**Pull request closed without merging**